java/java.openjdk.project/src/org/netbeans/modules/java/openjdk/project/ClassPathProviderImpl.java

```diff
@@ -203,7 +203,7 @@ private static URL projectDir2FakeTarget(FileObject projectDir) throws Malformed
     @Override
     public ClassPath findClassPath(FileObject file, String type) {
         if (sourceCP.findOwnerRoot(file) != null) {
-            if (!repository.isAnyProjectOpened()) {
+            if (repository != null && !repository.isAnyProjectOpened()) {
                 //if no project is open, java.base may not be indexed. Fallback on default queries:
                 return null;
             }
```

Comment:
comment might need adjustment to:
`//if no project is open or the JDK is not modular, java.base may not be indexed. Fallback on default queries:`

I assume repository is always null for openjdk 8 checkouts. cc @jlahoda

Reply:
For openjdk 8 (and maybe prior) checkouts, the repository is null so the comment inside that block wouldn't apply. Would non-modular JDKs ever have a non-null repository?

Reply:
IIRC this was an attempt to avoid unnecessary parsing from source when none of the OpenJDK project is open and hence not indexed. There's probably nothing better we could do for JDK 8.